### PR TITLE
add config and comment

### PR DESCRIPTION
**Pull request by regmibijay**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Tests added and passed if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://github.com/gventuri/pandas-ai/blob/main/CONTRIBUTING.md#-testing).

Currently `Queries` section of README.md passes kwargs directly using `config={"llm": llm}` which can be approached better with included configuration schema `Config` from `pandasai.schemas.df_config` and this serves more object-orientted approach. 

Added some minor code comments explaining already blatantly obvious steps.  

### Summary by CodeRabbit

---
- New Feature: Introduced a more flexible configuration for the `SmartDataframe` object. Users can now leverage the `Config` class from the `pandasai.schemas.df_config` module to customize their `SmartDataframe` instances. This enhancement provides users with greater control and adaptability when working with dataframes in the system.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ For example, you can ask PandasAI to find all the rows in a DataFrame where the
 ```python
 import pandas as pd
 from pandasai import SmartDataframe
+from pandasai.schemas.df_config import Config
 
 # Sample DataFrame
 df = pd.DataFrame({
```
[the remark this answers is not in the document]
Comment (regmibijay) [reply README.md:50]: Indenting was double checked and is proper.

**Review by mspronesti — COMMENTED** [review]
From what I've understood from this discussion with @gventuri [here](https://github.com/gventuri/pandas-ai/pull/512#discussion_r1311271167), the use of the `pandasai.schemas.df_config.Config` is not advised.

**Comment by gventuri** [comment]
@regmibijay @mspronesti I don't know, pushing too much to OOP seems verbose to me in this case. The idea is to keep instantiating a SmartDataFrame as easy as possible, and I cannot think of any real improvement in pushing the OOP approach rather than the other.

That said, I'm super oper for feedbacks, so let's have a conversation about it, if you want!

**Comment by regmibijay** [comment]
I personally would favour `Config` (or even better something like `SmartDataframeConfiguration`) because of following reasons: 

- It is easier to keep track of parameters supported by backend. I assume it has been experimented enough with pydantic validations before and you decided against it but I would love to invest some time in this regard so users have easier time understanding what is supported and how (and in which manner) parameters should be passed

- It is easier to create tests. We could write scenarios which define which combination of parameters is supported and can be provided each other. I think this would make sense because we have different set of `llm` instances which support different configurations. 

- Would make easier to refine arguments. For example `api_key_path` could be read from file for all instances of LLM and not only for openai. 

I see this requires significant effort and the discussion up until has been which solutions to use but there could also be the option of pass either in kwargs form or in Object form.

**Comment by gventuri** [comment]
Hi @regmibijay,

- I agree `Config` can be very generic and I don't mind `SmartDataframeConfiguration`, but it's a little bit "long" and has the problem that also the `SmartDatalake` shares the same config.
- I'm not sure I understand the advantage of passing it as kwargs as opposed to a dictionary. Can you explain? I think a hybrid approach of either a dict or a Config object could work. What do you think?
- I'm not sure I understand the point about LLMs. Every LLM has its own config variables as you instantiate it (or can read the default from `pandasai.json` file. The configs that are passed to the dataframe are different, including, for example, an instance of an LLM.

Happy to hear your feedback and even jumping on a call, if you prefer!

**Comment by regmibijay** [comment]
Hi,
thanks for your feedback and your thoughts.

- I meant to add that each of `SmartDatalake` and `SmartDataframe` could have each set of configuration classes, which could share same Abstract Base Class.  This could also make it possible to implement methods such as `SmartBaseConfiguration.from_dict()` which supports Kwargs. Bit like [this](https://huggingface.co/transformers/v2.10.0/main_classes/configuration.html) config class for transformers.

- The main problem I see with passing kwargs is that it feels a bit hacky and there is no way of knowing which options are available. With classes and docstrings, it is possible to either view Configuration signature in IDE or with `help` or in case of Juypter Notebooks with `SmartDFConfig?` 

- With 3rd point, I meant perhaps it could be easier to aggregate basic sets of parameters and config options for all LLMs. This will in return make writing tests bit easier in some cases.

**Comment by regmibijay** [comment]
Since there was no resonance, closing this PR for now.

**Pull request closed without merging**